File: LC202_HappyNumber.cpp
class Solution{
public:
    bool isHappy(int n)
    {
        vector<int> result;
        int val = getSquareSum(n);
        while(val != 1)
        {
            if(std::find(result.begin(), result.end(), val) != result.end())
                return false;
            result.push_back(val);
            val = getSquareSum(val);
        }
        reurn true;
    }

    int getSquareSum(int n)
    {
        int sum = 0, temp;
        while(n)
        {
            temp = n%10;
            sum += temp*temp;
            n = n / 10;
        }
        return sum;
    }
}

File: LC202_HappyNumber.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

class Solution:
    def isHappy(self, n):
        """
        type n: int
        rtype: bool
        """
        slow = self.getSquareSum(n)
        fast = self.getSquareSum(n)
        fast = self.getSquareSum(fast)
        while(slow != fast):
            slow = self.getSquareSum(slow)
            fast = self.getSquareSum(fast)
            fast = self.getSquareSum(fast)
        if( slow == 1 ):
            return True
        else:
            return False

    def getSquareSum(self, n):
        sum = 0
        while(n):
            tmp = n % 10
            sum += tmp*tmp
            n = n // 10

        return sum
